### PR TITLE
Add collaborators

### PR DESCRIPTION
**Pull request by MarioRodrigues10**

### CODE REVIEW

File: lib/atomic/accounts/user.ex [hunk 1]
```diff
@@ -15,8 +15,8 @@ defmodule Atomic.Accounts.User do
     field :confirmed_at, :naive_datetime
 
     field :role, Ecto.Enum, values: @roles
-
     has_many :enrollments, Enrollment
+    has_one :collaborator, Collaborator, on_replace: :delete_if_exists, on_delete: :delete_all
```
Comment (ruioliveira02) [inline lib/atomic/accounts/user.ex:19]:
```suggestion
    has_many :collaborator, Collaborator, on_replace: :delete_if_exists, on_delete: :delete_all
```
A user can be a collaborator of multiple departments

File: test/support/fixtures/accounts_fixtures.ex [hunk 2]
```diff
@@ -28,4 +28,16 @@ defmodule Atomic.AccountsFixtures do
...
+  Generate a collaborator.
+  """
+  def collaborator_fixture(attrs \\ %{}) do
+    {:ok, collaborator} =
+      attrs
+      |> Enum.into(%{})
+      |> Atomic.Accounts.create_collaborator()
+
+    collaborator
+  end
```
Comment (ruioliveira02) [inline test/support/fixtures/accounts_fixtures.ex:42]: We should use factories for testing (not saying to do it here, just mentioning in general)

**Review by ruioliveira02 — CHANGES_REQUESTED** [review]

File: lib/atomic/accounts/collaborator_department.ex [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    An activity speaker
+  """
+  use Atomic.Schema
+
+  alias Atomic.Departments.Department
+  alias Atomic.Accounts.Collaborator
+
+  @required_fields ~w(collaborator_id department_id)a
+
+  schema "collaborator_departments" do
```
Comment (feliciofilipe) [inline lib/atomic/accounts/collaborator_department.ex:12]: Should we not have a role to identify this relation, e.g.: 'President', 'Vice-President', etc?

File: lib/atomic/accounts/user.ex (hunk 1 above)
Comment (feliciofilipe) [reply lib/atomic/accounts/user.ex:19]: collaborators*

File: lib/atomic/accounts/collaborator_department.ex (hunk 3 above)
Comment (MarioRodrigues10) [reply lib/atomic/accounts/collaborator_department.ex:12]: I thought about creating another schema "board_users" where i give those roles to an user

**Comment by feliciofilipe** [comment]
We need to discuss this carefully before going foward

File: lib/atomic/accounts/user.ex (hunk 1 above)
Comment (MarioRodrigues10) [reply lib/atomic/accounts/user.ex]: I updated to a many_to_many relationship

File: lib/atomic/accounts/collaborator_department.ex (hunk 3 above)
Comment (MarioRodrigues10) [reply lib/atomic/accounts/collaborator_department.ex:12]: Inside this schema i can create ```  role field ``` and give it to user , what you think?

File: lib/atomic/accounts/collaborator_department.ex [hunk 4]
```diff
@@ -0,0 +1,29 @@
+defmodule Atomic.Accounts.CollaboratorDepartment do
+  @moduledoc """
+    An activity speaker
```
Comment (ruilopesm) [inline lib/atomic/accounts/collaborator_department.ex]:
```suggestion
    A relation representing a collaborator who belongs to a department
```

File: priv/repo/migrations/20230114142351_create_collaborator_departments.exs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+defmodule Atomic.Repo.Migrations.CreateCollaboratorDepartments do
+  use Ecto.Migration
+
+  def change do
+    create table(:collaborator_departments, primary_key: false) do
+      add :id, :binary_id, primary_key: true
+
+      add :collaborator_id, references(:users, on_delete: :nothing, type: :binary_id)
```
Comment (ruioliveira02) [inline priv/repo/migrations/20230114142351_create_collaborator_departments.exs:8]:
```suggestion
      add :user_id, references(:users, on_delete: :nothing, type: :binary_id)
```

File: lib/atomic/accounts/collaborator_department.ex [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  """
+  use Atomic.Schema
+
+  alias Atomic.Departments.Department
+  alias Atomic.Accounts.User
+
+  @required_fields ~w(collaborator_id department_id)a
+
+  schema "collaborator_departments" do
+    belongs_to :collaborator, User
```
Comment (ruioliveira02) [inline lib/atomic/accounts/collaborator_department.ex:13]:
```suggestion
    belongs_to :user, User
```

**Review by ruioliveira02 — COMMENTED** [review]
As collaborators change every year you should add the year as a field IMO

**Comment by MarioRodrigues10** [comment]
This was closed because it was done in #291

**Pull request closed without merging**